exercises/matrix/src/test/java/MatrixTest.java

```diff
@@ -0,0 +1,184 @@
+
+        @Parameterized.Parameters(name = "{index}: checking last row of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0, 1}},
+                    {"0\\n1", new int[] {1}},
+                    {"0 1\\n2 3", new int[] {2, 3}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {6, 7, 8}}
+            });
+        }
+
+        public GetLastRowTest(String matrixAsString, int[] lastRow) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.lastRow = lastRow;
+        }
+
+        @Ignore
+        @Test
+        public void getLastRowTest() {
+            assertArrayEquals(matrix.getRow(matrix.getRowsCount() - 1), lastRow);
+        }
+    }
+
+    @RunWith(Parameterized.class)
+    public static class GetFirstColumnTest {
+        private Matrix matrix;
+        private int[] firstColumn;
+
+        @Parameterized.Parameters(name = "{index}: checking first column of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0}},
+                    {"0\\n1", new int[] {0, 1}},
+                    {"0 1\\n2 3", new int[] {0, 2}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {0, 3, 6}}
+            });
+        }
+
+        public GetFirstColumnTest(String matrixAsString, int[] firstColumn) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.firstColumn = firstColumn;
+        }
+
+        @Ignore
+        @Test
+        public void getFirstColumnTest() {
+            assertArrayEquals(matrix.getColumn(0), firstColumn);
+        }
+    }
+
+
+    @RunWith(Parameterized.class)
+    public static class GetLastColumnTest {
+        private Matrix matrix;
+        private int[] lastColumn;
+
+        @Parameterized.Parameters(name = "{index}: checking last column of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {1}},
+                    {"0\\n1", new int[] {0, 1}},
+                    {"0 1\\n2 3", new int[] {1, 3}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {2, 5, 8}}
+            });
+        }
+
+        public GetLastColumnTest(String matrixAsString, int[] lastColumn) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.lastColumn = lastColumn;
+        }
+
+        @Ignore
+        @Test
+        public void getLastColumnTest() {
+            assertArrayEquals(matrix.getColumn(matrix.getColumnsCount() - 1), lastColumn);
```

Comment:
JUnit assertions typically request the expected value before the actual value, so we should reverse the parameter ordering of all of these calls to `assertArrayEquals` - otherwise our failure messages will be backwards and potentially confusing!